opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -67,14 +67,33 @@ def record(self, value: ValueT) -> None:
         """
 
 
+class LabelSet:
+    """A canonicalized set of labels useful for preaggregation
+
+    Re-usable LabelSet objects provide a potential optimization for scenarios
+    where handles might not be effective. For example, if the LabelSet will be
+    re-used but only used once per metrics, handles do not offer any
+    optimization. It may best to pre-compute a canonicalized LabelSet once and
+    re-use it with the direct calling convention. LabelSets are immutable and
+    should be opaque in implementation.
+    """
+
+
+class DefaultLabelSet(LabelSet):
```

Comment:
Just like how `metric` and `handle` have default implementations, we don't want to have the `meter` functions to return None objects (it would be `get_label_set` in this case). Also a LabelSet has no methods so I didn't think it was necessary to use an ABC (unless my understanding of the benefits is mistaken).

Reply:
My point here is that since `LabelSet` is not an ABC we technically don't need `DefaultLabelSet` since we could return a plain `LabelSet` instance from `Meter.get_label_set`.